php/php.project/src/org/netbeans/modules/php/project/ComputeTestMethodAnnotations.java

```diff
@@ -142,59 +178,72 @@ public void run() {
         }
     }
 
-    private List<TestMethod> getTestMethods(Document document) {
-        List<TestMethod> testMethods = new ArrayList<>();
-        FileObject fileObject = NbEditorUtilities.getFileObject(document);
-        if (fileObject != null) {
-            PhpProject project = PhpProjectUtils.getPhpProject(fileObject);
-            assert project != null;
-            PhpModule phpModule = project.getPhpModule();
-            for (PhpTestingProvider testingProvider : project.getTestingProviders()) {
+    /**
+     * Get test methods for the first available testing provider.
+     *
+     * @return test methods
+     */
+    private List<TestMethod> getTestMethods() {
```

Comment:
I am not sure about this change, the previous code made more sense to me. The reason is the "lifecycle" - I mean, we detect a possible document to process, so we schedule processing for this particular document. If the focused document already changed, we stop the processing and schedule a new one for this new document. 

With this change, the `handledDocument` can change, no? Or is it ensured that we are using the very _same_ instance on line 187 and 195?

Reply:
You are right! I'm wrong, sorry... Will fix it. (will add `Document` to the parameter)

Reply:
No problem, great that we have the same understanding.

Reply:
Fixed: https://github.com/apache/netbeans/compare/eb40f846e45be782f35cf3bde9e0a93ec5544309..b4fcdad4623277619ec2dd6aca98aaf6b2b83bcf

Reply:
Looks good to me!